universal-login-contracts/contracts/ERC1077.sol

```diff
@@ -128,4 +131,13 @@ contract ERC1077 is KeyHolder, IERC1077 {
             msg.sender.transfer(gasUsed.mul(gasPrice));
         }
     }
+
+    function isSignerValid(address signer, address[] memory signers) private view returns (bool) {
+        for (uint i = 0;i < signers.length; i++) {
```

Comment:
This is quite expensive, we can avoid O(N^2) comparisons by doing a little trick:
- always sort signers in the same order (in javascript as they are passed as signatures, requires a helper js function)
- require each signer to be different than previous
- require each signer to be >= then previous (you can compare addresses same way as integers in solidity)